Frontend/ui-customer/src/Flow.purs

```diff
@@ -375,6 +378,43 @@ chooseLanguageScreenFlow = do
                             enterMobileNumberScreenFlow
     Refresh state -> chooseLanguageScreenFlow
 
+updateCustomerVersion :: Maybe Version -> Maybe Version -> FlowBT String Unit
+updateCustomerVersion dbClientVersion dbBundleVersion = do
+  let versionName = getValueToLocalStore VERSION_NAME
+      bundle = getValueToLocalStore BUNDLE_VERSION
+      clientVersionArray = (split (Pattern ".") versionName)
+      majorClientVersion = fromMaybe 1 (INT.fromString $ fromMaybe "NA" $ clientVersionArray !! 0)
+      minorClientVersion = fromMaybe 1 (INT.fromString $ fromMaybe "NA" $ clientVersionArray !! 1)
+      patchClientVersion = fromMaybe 1 (INT.fromString $ fromMaybe "NA" $ clientVersionArray !! 2)
+      bundleVersionArray = (split (Pattern ".") bundle)
+      majorBundleVersion = fromMaybe 1 (INT.fromString $ fromMaybe "NA" $ bundleVersionArray !! 0)
+      minorBundleVersion = fromMaybe 1 (INT.fromString $ fromMaybe "NA" $ bundleVersionArray !! 1)
+      patchBundleVersion = fromMaybe 1 (INT.fromString $ fromMaybe "NA" $ bundleVersionArray !! 2)
+
+  let (Version clientVersion) = Version {
+    major : majorClientVersion,
+    minor : minorClientVersion,
+    maintenance : patchClientVersion
+  }
+  let (Version bundleVersion) = Version {
+    major : majorBundleVersion,
+    minor : minorBundleVersion,
+    maintenance : patchBundleVersion
+  }
+  let (Version bundleVersion') = fromMaybe (Version bundleVersion) dbBundleVersion
+      dbBundleMajor = bundleVersion'.major
+      dbBundleMinor = bundleVersion'.minor
+      dbBundleMaintainence = bundleVersion'.maintenance
+  let (Version clientVersion') = fromMaybe (Version clientVersion) dbClientVersion
+      dbclientMajor = clientVersion'.major
+      dbclientMinor = clientVersion'.minor
+      dbclientMaintainence = clientVersion'.maintenance
+      
+  if (dbBundleMinor /= bundleVersion.minor || dbBundleMajor /= bundleVersion.major || dbBundleMaintainence /= bundleVersion.maintenance || dbclientMinor /= clientVersion.minor || dbclientMajor /= clientVersion.major || dbclientMaintainence /= clientVersion.maintenance) then do
```

Comment:
you are creating variables just to compare values, instead you can directly compare `dbBundleMajor'` with `bundleVersion`